packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,342 @@
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+</details>
+
+<details>
+<summary><code>useSelect</code> hook</summary>
+
+#### useSelect
+
+```jsx
+const { selectItem, reset, dropdownRef } = useSelect();
+
+const onSomeAction = () => {
+  selectItem('mumbai');
+};
+
+const onResetClick = () => {
+  reset();
+};
+
+<Dropdown ref={dropdownRef}>
+  <SelectInput />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+Inspirations
+
+</details>
+
+## Concluded Solution
+
+Based on the [discussions](#approach-comparisons), we decided to go with a combination of approaches propsed above-
+
+- On `SelectInput`, `value` + `onChange` + `defaultValue`
+- On `ActionListItem`, `isSelected` + `onClick` + `isDefaultSelected`
+
+### When Trigger is `SelectInput`
+
+For SelectInput (and maybe AutoComplete in future), we decided to go with `value` + `onChange` props on the trigger. This keeps the controlled state props on the same component as opposed to `isSelected` + `onChange` approach where `isSelected` goes on ActionListItem and `onChange` goes on SelectInput.
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+const onResetClick = () => {
+  setCurrentSelection(undefined);
+};
+
+const onSelectChange = ({ values }) => {
+  setCurrentSelection(values[0]);
+};
+
+<Dropdown>
+  <SelectInput
+    label="City"
+    // New Prop 👇🏼
+    value={currentSelection}
+    onChange={onSelectChange}
+  />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+### When Trigger is Button or other triggers
+
+When it's Button or other trigger, we use `isSelected` and `onClick` from `ActionListItem`
+
+```jsx
+const [currentSelection, setCurrentSelection] = React.useState();
+const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+const onSomeAction = () => {
+  setCurrentSelection('mumbai');
+};
+
+<Dropdown>
+  <DropdownButton>Show Dropdown</DropdownButton>
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem
+        title="Mumbai"
+        value="mumbai"
+        // New Props 👇🏼
+        isSelected={currentSelection === 'mumbai'}
+        onClick={() => {
+          // This handles internal controlled state
+          setCurrentSelection('mumbai');
+        }}
+      />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
```

Comment:
I think a note on the functionality with selectionType is missing.

- what if selection type is single but user passes isSelected for multiple actionlistitems

Reply:
Added this and another note about other impossible state in additional notes in second last section